sdks/go/pkg/beam/core/graph/fn.go

```diff
@@ -43,50 +43,46 @@ type Fn struct {
 // Name returns the name of the function or struct.
 func (f *Fn) Name() string {
 	if f.Fn != nil {
-		return f.Fn.Name
+		return f.Fn.Fn.Name()
 	}
 	t := reflectx.SkipPtr(reflect.TypeOf(f.Recv))
 	return fmt.Sprintf("%v.%v", t.PkgPath(), t.Name())
 }
 
 // DynFn is a generator for dynamically-created functions:
 //
-//      gen:   []byte -> T
+//    gen: (name string, t reflect.Type, []byte) -> func : T
 //
-// where the generated function, fn : T, is created by reflect.MakeFunc and
-// uses reflect.Values. The type, T, must be a valid funcx.Fn type. The
-// generator takes some arbitrary data as a []byte input. This concept allows
-// serialization of dynamically-generated functions, which do not have a valid
-// (unique) symbol. All such functions use the "reflect.makeFuncStub" symbol.
+// where the generated function, fn : T, is re-created at runtime. This concept
+// allows serialization of dynamically-generated functions, which do not have a
+// valid (unique) symbol such as one created via reflect.MakeFunc.
 type DynFn struct {
-	// Name is the (fake) name of the function. The actual symbol name is
-	// always "reflect.makeFuncStub".
+	// Name is the name of the function. It does not have to be a valid symbol.
 	Name string
-	// T is the type of the function
+	// T is the type of the generated function
 	T reflect.Type
-	// Data holds the data for the generator.
+	// Data holds the data for the generator. This data is trivially serializable.
```

Comment:
Maybe we don't need the comment. What []byte isn't trivially serializable?

Or maybe this is a comment for DynFn, that they are trivially serializable because of this, which is why they are awesome, and we built them? :)

Reply:
Clarified the comment.